python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -1092,19 +1082,24 @@ def _initialize_adapter(self, args: Sequence[str]) -> BaseAdapter:
         return adapter
 
     def get_defer_args(self) -> Sequence[str]:
-        """Returns ('--defer', '--defer-state', state_artifacts_dir) if state_dir was
-        specified and contains a manifest.json, () otherwise.
+        """Build the defer arguments for the dbt CLI command, using the supplied state directory.
+        If no state directory is supplied, or the state directory does not have a manifest for.
+        comparison, an exception is raised.
+
+        Returns:
+            Sequence[str]: The defer arguements for the dbt CLI command.
         """
+        if not (self.state_dir and Path(self.state_dir).joinpath("manifest.json").exists()):
+            raise DagsterDbtManifestNotFoundError(
+                "A valid state directory with a manifest.json is required to build defer arguments."
+                f"The following state directory was provided: `{self.state_dir}`."
+            )
```

Comment:
Probably don't want to error here, since I realized the ideal API here is that `get_defer_args` will always be invoked, regardless of the presence of state.

This way, your dbt assets are defined the same way, when executing in a full deployment or branch deployment.

Still believe that we should relax the requirement that `manifest.json` is present, so removing that.

```python
@dbt_assets(manifest=project.manifest_path)
def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
    yield from dbt.cli(["build", *dbt.get_defer_args()], context=context).stream()
```